e2e/policy/policy_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+//go:build e2e
+
+package policy
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"os"
+	"path"
+	"slices"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/edgelesssys/contrast/e2e/internal/contrasttest"
+	"github.com/edgelesssys/contrast/e2e/internal/kubeclient"
+	"github.com/edgelesssys/contrast/internal/kubeapi"
+	"github.com/edgelesssys/contrast/internal/kuberesource"
+	"github.com/edgelesssys/contrast/internal/manifest"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	opensslBackend  = "openssl-backend"
+	opensslFrontend = "openssl-frontend"
+	coordinator     = "coordinator"
+)
+
+var (
+	imageReplacementsFile, namespaceFile string
+	skipUndeploy                         bool
+)
+
+func TestPolicy(t *testing.T) {
+	ct := contrasttest.New(t, imageReplacementsFile, namespaceFile, skipUndeploy)
+
+	resources := kuberesource.OpenSSL()
+
+	coordinatorBundle := kuberesource.CoordinatorBundle()
+	resources = append(resources, coordinatorBundle...)
+	resources = kuberesource.AddPortForwarders(resources)
+
+	ct.Init(t, resources)
+
+	// initial deployment with pod allowed
+	require.True(t, t.Run("generate", ct.Generate), "contrast generate needs to succeed for subsequent tests")
+
+	require.True(t, t.Run("apply", ct.Apply), "Kubernetes resources need to be applied for subsequent tests")
+
+	require.True(t, t.Run("set", ct.Set), "contrast set needs to succeed for subsequent tests")
+	require.True(t, t.Run("contrast verify", ct.Verify), "contrast verify needs to succeed for subsequent tests")
+
+	t.Run("pod cannot join after it was removed from the manifest", func(t *testing.T) {
+		require := require.New(t)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+		defer cancel()
+
+		c := kubeclient.NewForTest(t)
+
+		t.Log("Waiting for deployments")
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslFrontend))
+
+		time.Sleep(5 * time.Second) // let the error counter go up initially
```

Comment:
I did encounter some problems, so I added it to be sure that I give everything enough time. It could very well be that this issue was solved later on, so I can try removing it and test everything again, then see if I encounter problems again.